### ISSUE TITLE
Overlapping driving_distance analysis for many startpoint scenario

### ISSUE THREAD

**Issue by Styp** [pgRouting/pgrouting#217]
Right now driving_distance analysis is only calculation the points in reach from a single point with a given time.

It would be very benefitial to calculate overlapping driving_distance polygons for multiple points. Below I added an example, as I implemented the cleanup and points grouping functions in my little Java framework.

First driving distance calculation:
![driving_1](https://f.cloud.github.com/assets/1202069/1608515/f7b8a346-54f9-11e3-81ef-221173f1f5b4.png)

Another driving distance calculation:
![driving_2](https://f.cloud.github.com/assets/1202069/1608517/f7f5bbbe-54f9-11e3-895a-581c23a010d7.png)

And now lets see how they interact and which points are assigned to which start point
![driving_overlap](https://f.cloud.github.com/assets/1202069/1608516/f7e17ff0-54f9-11e3-8230-fded79d3a2d4.png)

And now a true multi one...
![driving_overlap_multi](https://f.cloud.github.com/assets/1202069/1608519/0b1ecd7a-54fa-11e3-97cf-296b5a683605.png)

I hope that the images explain the idea, imho this feature could be usefull for a lot of pgR users!

**Comment by cvvergara** [comment]
@Styp  
if 100, 200, 300, 400 are vertices id's
10000 is in meters
The query you want would look something like this?
select * from pgr_drivingDistance('select * from edge_table', array [100, 200, 300], 10000) 

and say that 
from 100 to 400 cost is 200
from 200 to 400 cost is 500
from 300 to 400 cost is 600
then the 400 vertex would belong to id1=100

Therefore the requested output would look like?
```
seq     id1       id2               id3      cost
1         100     vertexid          edgeid        100
2         100     400               edgeid        200
.....
1004      100     vertexid          edgeid       9999
1005      200     vertexid          edgeid         80
....
2000      200      vertexid         edgeid       9999
2001      300      vertexid         edgeid        125
...
3006      300      vertexid         edgeid       9999
```

**Comment by cvvergara** [comment]
@robe2 Hello Regina:
Thanks for setting up ksp-improve-boost. I am not working on that branch, you can disconnect it from the automatic build.
 Now I am working on pgr_drivingDistance and I am using the following branch in my fork:
dd-develop
I am making some changes, and I am in the point where I need to know if compilation in other systems than linux is ok.

**Comment by robe2** [comment]
@cvvergara Will set up later today and disconnect  the  old  one.

**Comment by robe2** [comment]
@cvvergara Okay setup here:

http://winnie.postgis.net:1500/job/pgRouting_matrix_branch_cvvergara_dd-develop/

It's failing at i think both the 32-bit and 64-bit runs at compile

```
In file included from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:31:0,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp: In member function 'Path Path::getSubpath(unsigned int) const':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:106:19: error: 'i' does not name a type
         for (auto i = path.begin(); i != path.begin() + j; ++i) result.push_back((*i));
                   ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:106:37: error: expected ';' before 'i'
         for (auto i = path.begin(); i != path.begin() + j; ++i) result.push_back((*i));
                                     ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:106:37: error: 'i' was not declared in this scope
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp: In function 'Path equi_cost(const Path&, const Path&)':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:7: warning: lambda expressions only available with -std=c++11 or -std=gnu++11 [enabled by default]
       });
       ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8: error: no matching function for call to 'sort(std::deque<path_element3>::iterator, std::deque<path_element3>::iterator, equi_cost(const Path&, const Path&)::__lambda0)'
       });
        ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8: note: candidates are:
In file included from c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/algorithm:62:0,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/move/algorithm.hpp:24,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/move/move.hpp:24,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/util.hpp:19,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/buckets.hpp:14,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/table.hpp:10,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/equivalent.hpp:14,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/unordered_set.hpp:17,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered_set.hpp:16,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/graph/adjacency_list.hpp:21,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:28,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/bits/stl_algo.h:5474:5: note: template<class _RAIter, class _Compare> void std::sort(_RAIter, _RAIter, _Compare)
     sort(_RandomAccessIterator __first, _RandomAccessIterator __last,
     ^
c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/bits/stl_algo.h:5474:5: note:   template argument deduction/substitution failed:
In file included from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:31:0,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp: In substitution of 'template<class _RAIter, class _Compare> void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = std::_Deque_iterator<path_element3, path_element3&, path_element3*>; _Compare = equi_cost(const Path&, const Path&)::__lambda0]':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8:   required from here
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8: error: template argument for 'template<class _RAIter, class _Compare> void std::sort(_RAIter, _RAIter, _Compare)' uses local type 'equi_cost(const Path&, const Path&)::__lambda0'
       });
        ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8: error:   trying to instantiate 'template<class _RAIter, class _Compare> void std::sort(_RAIter, _RAIter, _Compare)'
In file included from c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/algorithm:62:0,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/move/algorithm.hpp:24,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/move/move.hpp:24,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/util.hpp:19,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/buckets.hpp:14,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/table.hpp:10,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/detail/equivalent.hpp:14,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered/unordered_set.hpp:17,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/unordered_set.hpp:16,
                 from e:/jenkins/boost/rel-1_53_0w64gcc48/include/boost-1_53/boost/graph/adjacency_list.hpp:21,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:28,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/bits/stl_algo.h:5438:5: note: template<class _RAIter> void std::sort(_RAIter, _RAIter)
     sort(_RandomAccessIterator __first, _RandomAccessIterator __last)
     ^
c:/ming64gcc48/mingw64/x86_64-w64-mingw32/include/c++/bits/stl_algo.h:5438:5: note:   template argument deduction/substitution failed:
In file included from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:31:0,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:257:8: note:   candidate expects 2 arguments, 3 provided
       });
        ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:259:22: error: ISO C++ forbids declaration of 'e' with no type [-fpermissive]
     for (auto const &e : p2.path) {
                      ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:259:26: error: range-based 'for' loops are not allowed in C++98 mode
     for (auto const &e : p2.path) {
                          ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:260:12: error: 'pos' does not name a type
       auto pos = find_if(result.path.begin(), result.path.end(),
            ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:263:19: error: expected primary-expression before ')' token
                  });
                   ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:263:19: error: expected ';' before ')' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:264:11: error: 'pos' was not declared in this scope
       if (pos != result.path.end()) {
           ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:265:27: error: request for member 'cost' in 'e', which is of non-class type 'const int'
         if (pos->cost > e.cost) {
                           ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:269:27: error: no matching function for call to 'Path::push_back(const int&)'
         result.push_back(e);
                           ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:269:27: note: candidates are:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:44:10: note: void Path::push_back(pgr_path_element3_t)
     void push_back(pgr_path_element3_t data) {
          ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:44:10: note:   no known conversion for argument 1 from 'const int' to 'pgr_path_element3_t {aka path_element3}'
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:74:10: note: void Path::push_back(int, int64_t, int64_t, int64_t, float8)
     void push_back(
          ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:74:10: note:   candidate expects 5 arguments, 1 provided
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp: In function 'Path equi_cost(const std::deque<Path>&)':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:277:22: error: ISO C++ forbids declaration of 'p1' with no type [-fpermissive]
     for (const auto &p1 : paths) {
                      ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:277:27: error: range-based 'for' loops are not allowed in C++98 mode
     for (const auto &p1 : paths) {
                           ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:278:36: error: no matching function for call to 'equi_cost(Path&, const int&)'
       result = equi_cost(result, p1);
                                    ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:278:36: note: candidates are:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:275:15: note: Path equi_cost(const std::deque<Path>&)
   friend Path equi_cost(const std::deque< Path > &paths) {
               ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:275:15: note:   candidate expects 1 argument, 2 provided
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:252:15: note: Path equi_cost(const Path&, const Path&)
   friend Path equi_cost(const Path &p1, const Path &p2) {
               ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/basePath_SSEC.hpp:252:15: note:   no known conversion for argument 2 from 'const int' to 'const Path&'
In file included from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:32:0,
                 from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp: In member function 'void Pgr_base_graph<G>::get_path(std::deque<Path>&, std::set<typename boost::graph_traits<Graph>::vertex_descriptor>, Pgr_base_graph<G>::V&) const':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:318:23: error: 'source' does not name a type
       for (const auto source: sources) {
                       ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected ';' before '}' token
     }
     ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected primary-expression before '}' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected ';' before '}' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected primary-expression before '}' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected ')' before '}' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected primary-expression before '}' token
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/././../../common/src/baseGraph.hpp:323:5: error: expected ';' before '}' token
In file included from e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/dijkstra_driver.cpp:32:0:
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp: In member function 'void Pgr_dijkstra<G>::dijkstra_dd(std::deque<Path>&, std::vector<long long int>, float8)':
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:178:24: error: ISO C++ forbids declaration of 'v_source' with no type [-fpermissive]
       for (const auto &v_source: v_sources) {
                        ^
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:178:34: error: range-based 'for' loops are not allowed in C++98 mode
       for (const auto &v_source: v_sources) {
                                  ^

src/dijkstra/src/CMakeFiles/dijkstra.dir/build.make:109: recipe for target `src/dijkstra/src/CMakeFiles/dijkstra.dir/dijkstra_driver.cpp.obj' failed
make[2]: *** [src/dijkstra/src/CMakeFiles/dijkstra.dir/dijkstra_driver.cpp.obj] Error 1
```

**Comment by cvvergara** [comment]
@robe2 Regina, 
Thanks, it will be failing, I am trying to use C++11 :(
e:/jenkins/pgrouting/branches/dd-develop/src/dijkstra/src/./pgr_dijkstra.hpp:178:34: error: range-based 'for' loops are not allowed in C++98 mode
       for (const auto &v_source: v_sources)

**Comment by cvvergara** [comment]
@robe2 
It compiled, but the test was not executed because of 'pgr_test__db__test'
RUNNING: test-runner.pl -pgisver 2.1.7 -pgport 10444 -ignorenotice
ERROR: test database 'pgr_test__db__test' exists, you must drop or rename it!

**Comment by cvvergara** [comment]
This feature was added on V2.1
I'll mark it as fixed on develop (V2.1 milestone is closed)
